pallets/genetic-testing/src/lib.rs

```diff
@@ -155,10 +155,10 @@ pub mod pallet {
         // ----------------------- Update --------------------------
 
         #[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
-        pub fn process_dna_sample(origin: OriginFor<T>, tracking_id: Vec<u8>) -> DispatchResultWithPostInfo {
+        pub fn process_dna_sample(origin: OriginFor<T>, tracking_id: Vec<u8>, status: DnaSampleStatus) -> DispatchResultWithPostInfo {
             let who = ensure_signed(origin)?;
 
-            match <Self as GeneticTestingInterface<T>>::process_dna_sample(&who, &tracking_id) {
+            match <Self as GeneticTestingInterface<T>>::process_dna_sample(&who, &tracking_id, status) {
                 Ok(dna_sample) => {
```

Comment:
```suggestion
                Ok(dna_sample) => {
                      match status {
                          DnaSampleStatus::Registered => Self::deposit_event(Event::<T>::DnaSampleRegistered(dna_sample.clone())),
                          DnaSampleStatus::Received => Self::deposit_event(Event::<T>::DnaSampleReceived(dna_sample.clone())),
                          ... etc
                      }
                }
```

Reply:
Yang ini spertinya kita harus emit event nya sesuai dengan status yang di update bli @SHA888

Reply:
Okay fixing it. Thank you mas @kevinjanada

Reply:
Jadinya udah aku tambahin mas @kevinjanada kalau sempat dicek lagi ya. Thank you

Reply:
Ini aku cek dulu deh.

Reply:
bli, sorry ini maksudku harusnya bisa pake match untuk emit event sesuai dengan DnaSampleStatus nya.